contracts/fast/FastTokenFacet.sol

```diff
@@ -137,7 +134,7 @@ contract FastTokenFacet is AFastFacet, IERC20 {
     // Since the holder's account is now empty, make sure to keep track of it both
     // in this FAST and in the marketplace.
     s.tokenHolders.remove(holder, true);
-    MarketplaceTokenHoldersFacet(LibFast.data().marketplace).fastBalanceChanged(holder, 0);
+    MarketplaceTokenHoldersFacet(LibFast.data().marketplace).fastBalanceChanged(holder, amount);
```

Comment:
I don't think this is right... or I might be misunderstanding...

We want the token holding - the original holder to be 'untracked' at the Marketplace level. However, we've just taken the balance (pre assigning the tokens to the 0 address) and passed it here... which means it's going to have an incorrect balance tracked at the Marketplace level.